prov/efa/src/efa_prov_info.c

```diff
@@ -740,3 +740,23 @@ int efa_prov_info_compare_domain_name(const struct fi_info *hints,
 
        return 0;
 }
+
+/*
+ * @brief Compare the bus id specified via hints and match it with the
+ *		  bus_id in prov_info
+ *
+ * @param      info[in]        info object
+ * @param      hints[in]       hints from user's call to fi_getinfo()
+ *
+ * return      1 - If the names are different
+ *             0 - No difference, names match.
+ */
+int efa_prov_info_compare_pci_bus_id(const struct fi_info *hints,
+                                     const struct fi_info *info)
+{
+	if (hints && hints->nic && hints->nic->bus_attr && hints->nic->bus_attr->bus_type == FI_BUS_PCI) {
+		return (hints->nic->bus_attr->attr.pci.bus_id == info->nic->bus_attr->attr.pci.bus_id) ? 0 : 1;
+	}
+
+	return 0;
```

Comment:
if `hints->nic->bus_attr->bus_type` is NULL this returns 0. Is this correct?

Reply:
`hints->nic->bus_attr->bus_type` is an integer.

I think you meant `hints->nic->bus_attr`. Yes, when it is null, we want to return 0. It means user did not specify a bus ID.